### PR TITLE
New website

### PR DESCRIPTION
**Pull request by amacneil**
1. Migrated MCAP website from mkdocs to docusaurus
   - Consistent with what we are using to generate Foxglove API docs
   - Ability to easily integrate React components, with full support for server side rendering
   - Foxglove team has much more familiarity with TypeScript/React than Python
   - Cleaner URLs

2. Migrated hosting from GitHub Pages to Cloudflare Pages
   - Supports PR preview deployments

3. Removed self-hosted rust docs in favor of https://docs.rs/mcap/
   - Easier to show the latest released version (not main) and browse docs from previous versions
   - Same pattern we use for Go docs

5. Updated copy
   - New homepage
   - Integrated mcap CLI docs into website
   - Fixed links and formatting due to website migration

Preview deployment here: https://adrian-website.mcap.pages.dev/

### CODE REVIEW

File: website/package.json
Comment (defunctzombie) [inline website/package.json:17]: Wonder why we pin these exact dependencies? BTW there's version 2.4.1 now

File: .github/workflows/website.yml
Comment (defunctzombie) [inline .github/workflows/website.yml:31]: What builds this thing?

File: .github/workflows/website.yml
Comment (defunctzombie) [inline .github/workflows/website.yml:128]: What deploys this to production once it merges to main?

File: .github/workflows/website.yml
Comment (amacneil) [reply .github/workflows/website.yml:31]: `yarn typedoc` immediate above

File: .github/workflows/website.yml
Comment (amacneil) [reply .github/workflows/website.yml:128]: I'm not super happy with this but would like to figure it out in a separate PR. This whole `environment` section is not needed and the only thing it seems to do is make the "view deployment" button appear on the PR. The actual deployment happens regardless.

I think probably the upstream cloudflare action is missing something to cause the PR deployment status to show up.

Comment (amacneil) [reply .github/workflows/website.yml:128]: Also the `deploy` step has an output called `steps.deploy.outputs.environment` with the environment name but Github does not let you use variables in this `name:` field.

Anyway like I said this line is purely cosmetic and I don't think should block merging the PR.

File: website/package.json
Comment (amacneil) [reply website/package.json:17]: what do you mean? we pin everything.

I guess they could move to `devDependencies` since this isn't a library

File: .github/workflows/website.yml
Comment (amacneil) [reply .github/workflows/website.yml]: Turns out this is fixed in the latest version of the pages action.

File: website/package.json
Comment (amacneil) [reply website/package.json]: updated to v2.4.1

File: .github/workflows/website.yml
Comment (amacneil) [reply .github/workflows/website.yml]: removed this line entirely by having typedoc output directly into the correct directory

File: website/src/pages/index.tsx
Comment (jtbandes) [inline website/src/pages/index.tsx:30]:
```suggestion
    title: "High-performance writing",
```

File: website/src/pages/index.tsx
Comment (jtbandes) [inline website/src/pages/index.tsx:45]:
```suggestion
      "MCAP files support optional indexing, allowing for fast, efficient reading, even over a low-bandwidth internet connection.",
```

File: website/src/pages/index.tsx
Comment (jtbandes) [inline website/src/pages/index.tsx:57]:
```suggestion
      "MCAP reader and writer libraries are available in many languages, including C++, Go, Python, Rust, Swift, and Typescript.",
```

File: website/src/pages/index.tsx
Comment (jtbandes) [inline website/src/pages/index.tsx:66]:
```suggestion
    title: "Production-grade",
```

File: website/src/pages/index.tsx
Comment (jtbandes) [inline website/src/pages/index.tsx:116]: What's "container"? Is that a docusaurus stock style?

File: website/src/pages/index.tsx
Comment (jtbandes) [inline website/src/pages/index.tsx:15]: These features use inconsistent parts of speech, e.g. "self-contained"/"multilingual" vs "pub/sub logging".

**Comment by jtbandes** [comment]
Blue, purple, and green is a pretty weird color combo:
image

File: website/docs/reference/index.md
Comment (jtbandes) [inline website/docs/reference/index.md:26]: These footnotes seem to show up with the wrong numbers on the page.

Old site:
image

New site:
image

File: website/docs/guides/cli.md
Comment (jtbandes) [inline website/docs/guides/cli.md:30]: should we just delete this section then?

**Review by jtbandes — COMMENTED** [review]
It's not great that these link descriptions get cut off:

image

File: website/src/pages/index.tsx
Comment (amacneil) [reply website/src/pages/index.tsx:30]: is that correct? why would you hyphenate it?

File: website/src/pages/index.tsx
Comment (amacneil) [reply website/src/pages/index.tsx:45]: "indexing" sounds like something I would do external to the file (e.g. Data Platform "indexes" mcap files), versus "mcap files contain an index". don't you think?

File: website/src/pages/index.tsx
Comment (amacneil) [reply website/src/pages/index.tsx:66]: why hyphenate? these seem like separate words

File: website/src/pages/index.tsx
Comment (amacneil) [reply website/src/pages/index.tsx:116]: Yeah it's some docusaurus thing. I the only real benefit we get is making sure the homepage content has the same max-width and margin as the footer content.

File: website/src/pages/index.tsx
Comment (amacneil) [reply website/src/pages/index.tsx:15]: Sorry not sure what you are trying to say here

File: website/docs/guides/cli.md
Comment (amacneil) [reply website/docs/guides/cli.md:30]: I think it could be helpful having something about how to install from latest main? Honestly I have no idea why this is "not recommended", I moved this text from the existing CLI readme.

File: website/docs/guides/index.md
Comment (amacneil) [reply website/docs/guides/index.md:9]: Agreed, I was thinking we should clean this up in a separate PR. I could take a look in this one but the PR is already so massive and it would be easier to review copy changes in smaller chunks.

File: website/src/pages/index.tsx
Comment (jtbandes) [reply website/src/pages/index.tsx:30]: https://www.grammarly.com/blog/hyphen-with-compound-modifiers/

File: website/src/pages/index.tsx
Comment (jtbandes) [reply website/src/pages/index.tsx:66]: https://www.grammarly.com/blog/hyphen-with-compound-modifiers/

File: website/src/pages/index.tsx
Comment (jtbandes) [reply website/src/pages/index.tsx:15]: @esthersweon might have a better way of explaining since I think this parallels what we do for blog post titles.

Comment (jtbandes) [reply website/src/pages/index.tsx:15]: It reads like this:
- MCAP is Serialization-agnostic ✅ 
- MCAP is Self-contained ✅ 
- MCAP is Multilingual ✅ 
- MCAP is Production-grade ✅ 
- MCAP is Optional compression ❓
- MCAP is High-performance writing ❓

**Comment by amacneil** [comment]
@jtbandes since Github doesn't support threading on regular comments:

>  not great that these link descriptions get cut off 

Agreed. The "category" pages could use some more work. Doesn't seem like a blocker to merging since they are redundant with the sidebar navigation anyway.

> These ["MCAP" text and logo] seem too far apart IMO

Ok I'll move them closer together. I'm hoping we do another pass at improving this whole homepage design in a follow up PR.

> Blue, purple, and green is a pretty weird color combo

Yeah, it's not my best work. At least it's only when you hover that you see all 3. If you have a specific color suggestion feel free to make it.

Again, we have some designs floating around for homepage improvements, but I would like to get the architectural change landed so that other PRs can be more targeted.

**Comment by jtbandes** [comment]
>  If you have a specific color suggestion feel free to make it.

Can we replace all the green with something not green? e.g. purple.

**Comment by james-rms** [comment]
:+1: I think the CLI docs seem out of place under guides, I'd put them as another heading in the topnav.

File: website/docs/reference/index.md
Comment (amacneil) [reply website/docs/reference/index.md:26]: 🤔

File: website/src/pages/index.tsx
Comment (jtbandes) [reply website/src/pages/index.tsx:45]: feel free to ignore this if you don't like it

File: website/src/pages/index.tsx
Comment (amacneil) [reply website/src/pages/index.tsx:30]: TIL

File: website/src/pages/index.tsx
Comment (amacneil) [reply website/src/pages/index.tsx:66]: Ok I accepted your other suggestion, but I still disagree with this one because it is not clearly an adjective in a sentence.

E.g. `production-grade software` seems correct to hyphenate but `mcap is production grade` should not have a hyphen.

https://www.stcloudstate.edu/writeplace/_files/documents/grammar-punctuation/hyphen-rules.pdf

Comment (jtbandes) [reply website/src/pages/index.tsx:66]: yeah I guess I applied the rule arbitrarily in my head because it's a sentence fragment.

On the other hand, I don't know actually know if the rule applies to things like this where the second word of the pair is a noun. For example, `top-tier` is just a word in the dictionary: https://www.merriam-webster.com/dictionary/top-tier It's not acting like two separate words that happen to be hyphenated 🤷   In my mind `production-grade` should follow the same rule as `top-tier` but I can't really back that up with sources.

File: website/docs/reference/index.md
Comment (amacneil) [reply website/docs/reference/index.md:26]: https://github.com/facebook/docusaurus/issues/6625

**Comment by jtbandes** [comment]
still got some green
image

File: website/docs/reference/index.md
Comment (amacneil) [reply website/docs/reference/index.md:26]: worked around by using manual `<sup>` instead of markdown footnotes

**Comment by amacneil** [comment]
> 👍 I think the CLI docs seem out of place under guides, I'd put them as another heading in the topnav.

Yeah I tried this, it felt pretty lonely (e.g. left sidebar only has one item). If the CLI docs were fleshed out with multiple pages of content I think it would work. Could always do that later.

**Comment by amacneil** [comment]
> still got some green

I decided to embrace the green and remove the purple. It looks decent for now and we can always change colors in follow up PR.

File: .github/workflows/ci.yml
Comment (jtbandes) [inline .github/workflows/ci.yml:169]: do we not run prettier --check in CI anymore?

File: .github/workflows/website.yml
Comment (jtbandes) [inline .github/workflows/website.yml:126]: Looks like they've released 1.5.0 now

**Review by jtbandes — APPROVED** [review]

**Comment by jtbandes** [comment]
Why Cloudflare Pages as opposed to GitHub pages or something else?

**Comment by amacneil** [comment]
> Why Cloudflare Pages as opposed to GitHub pages or something else?

Per PR description - "Supports PR preview deployments", that is the only reason. GitHub Pages seems to be an abandoned product, surprised they haven't shipped more features to become competitive with Vercel etc.

I did originally use Vercel for the deployment, but wasn't excited that they share a deploy key across all projects in the org which seemed non-ideal for security.

File: .github/workflows/website.yml
Comment (amacneil) [reply .github/workflows/website.yml:126]: 🎉

File: .github/workflows/ci.yml
Comment (amacneil) [reply .github/workflows/ci.yml:169]: Good catch, fixed.

In this PR I actually made prettier run on the entire repo, but forgot to add it to CI.

**Pull request merged**
